lib/galaxy/webapps/galaxy/api/workflows.py

```diff
@@ -91,20 +78,25 @@ def set_workflow_menu( self, trans, **kwd ):
         q = sess.query( model.StoredWorkflow )
         # To ensure id list is unique
         seen_workflow_ids = set()
-        for id in workflow_ids:
-            if id in seen_workflow_ids:
-                continue
-            else:
-                seen_workflow_ids.add( id )
-            m = model.StoredWorkflowMenuEntry()
-            m.stored_workflow = q.get( id )
-            user.stored_workflow_menu_entries.append( m )
+        for wf_id in workflow_ids:
```

Comment:
I'd probably just decode all workflow ids using `trans.security.decode_id` as early as you can. The rest of the code can stay the same then.